composer/cli/launcher.py

```diff
@@ -494,11 +501,13 @@ def main():
         # what failed. No need to re-raise the exception, as `aggregate_process_returncode`
         # will return an appropriate error code, which will cause the script to exit.
         traceback.print_exc()
-        print('Killing training processes')
     finally:
         _cleanup_processes(processes)
         log_tmpdir.cleanup()
-        return _aggregate_process_returncode(processes)
+        return_code = _aggregate_process_returncode(processes)
+        if args.kill_python_zombies:
```

Comment:
It looks like, as written, this happens whether `composer` finished successfully or not. Should we only do it if there is an error? I'm assuming we can tell something went wrong from the `return_code`

Reply:
FWIW even though this is optional I think it's a bit too extreme to live in the launcher.  This might make sense in the isolated context of the Mosaic Platform but I would be so annoyed if I enabled this locally and it killed all python processes on my system.

If we had a way of building the process tree under the `composer` root process and cleaning up or if we explicitly searched for processes without a parent (zombies) and only `pkill`'d those then this would be ok.